playbooks/robusta_playbooks/pod_troubleshooting.py

```diff
@@ -277,20 +292,47 @@ def debugger_stack_trace(event: PodEvent, params: DebuggerParams):
         failure=False,
     )
-    cmd = f"debug-toolkit stack-trace {pid}"
+    cmd = f"debug-toolkit stack-trace {pid} --amount={params.traces_amount} --sleep-duration-s={params.sleep_duration_s}"
     output = RobustaPod.exec_in_debugger_pod(
         pod.metadata.name,
         pod.spec.nodeName,
         cmd,
     )
     blocks = []
-    for thread_output in output.split("\n\n"):
-        if thread_output.startswith("Current thread"):
-            # this is the thread we are getting the stack trace from, not relevant for debugging
-            continue
-        if thread_output:
-            blocks.append(MarkdownBlock(f"```\n{thread_output}\n```"))
-    finding.add_enrichment(blocks, annotations={SlackAnnotations.ATTACHMENT: True})
+    try:
+        output_json = json.loads(output)
+        if len(output_json) == 0 or (len(output_json) == 1 and output_json[0]["status"].lower() != "success"):
+            # no stack traces returned or only one with error
+            error_message = 'Failed to get python stack trace'
+            if len(output_json) == 1:
+                error_message += f', debugger error {output_json[0]["error"]} at {output_json[0]["trace"]}'
+            logging.error(error_message)
+            blocks.append(MarkdownBlock(f"Error while getting python stack trace."))
+        elif len(output_json) == 1 and output_json[0]["status"] == "success":
+            # print single stack trace directly to finding
+            single_stack_trace = output_json[0]["trace"]
+            for thread_output in single_stack_trace.split("\n\n"):
+                if thread_output.startswith("Current thread"):
+                    # this is the thread we are getting the stack trace from, not relevant for debugging
+                    continue
+                if thread_output and output_json[0]["status"].lower() == "success":
+                    blocks.append(MarkdownBlock(f"```\n{thread_output}\n```"))
+        else:
+            # print multiple stack traces to file
+            clean_output = []
+            for trace_object in output_json:
+                if trace_object["status"] != "success":
+                    # the full python stack trace of the error will appear here
+                    logging.error(f'Failed to get stack trace, debugger error {trace_object["error"]} at {trace_object["trace"]}')
+                    clean_output.append({"time": trace_object["time"], "status": "Error: Failed to get stack trace."})
+                else:
+                    clean_output.append(trace_object)
+            clean_file_output = json.dumps(clean_output, indent=4, sort_keys=True).replace('\\n', '\n')
+            blocks.append(FileBlock(f"debugger_stack_trace_{pid}.txt", clean_file_output.encode()))
+    except ValueError:  # includes simplejson.decoder.JSONDecodeError, could still be valid output
+        logging.error(f"failed to decode output")
+        blocks.append(MarkdownBlock(f"```\n{output}\n```"))
```

Comment:
What is output here?
I don't think we should send, unexpected random data to the user